packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -155,13 +164,10 @@ export class SearchPage extends Component {
 
   private unsubscribeUrlManager!: Unsubscribe;
 
-  constructor(props: {}) {
-    super(props);
+  constructor(props: {}, context: AppContextType) {
+    super(props, context);
```

Comment:
Receiving the context from the constructor might be deprecated. Maybe we should test if this still works with the latest version of React.

Reply:
Indeed, I saw it was deprecated in he VSCode intellisense but did not find an alternative approach to access the context inside the constructor of a class component.

Accessing the context in the constructor is important because React does not execute the `componentDidMount` lifecycle method on the server. Maybe [static getStateDerivedFromProps](https://reactjs.org/docs/react-component.html#mounting) could be an alternative option,

Reply:
Unfortunately we should probably look into alternatives then, since I believe they may have removed the feature altogether from new versions of React.